src/compiler/parser.ts

```diff
@@ -4926,7 +4937,7 @@ namespace ts {
 
             const fullStart = getNodePos();
             const decorators = parseDecorators();
-            const modifiers = parseModifiers();
+            const modifiers = parseModifiers([SyntaxKind.ConstKeyword]);

```

Comment:
This means that every time you parse a class element, you are going to allocate an array. Furthermore, this is the only modifier that you are special-treating for now. I would suggest that you just pass a boolean flag for now called `permitConstAsModifier`